### PR TITLE
clingo 5.2.2 (new formula)

### PR DESCRIPTION
**Pull request by domoritz**
Replaces old Gringo formula. 

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/clingo.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Clingo < Formula
+  desc "ASP system to ground and solve logic programs"
+  homepage "https://potassco.org/"
+  url "https://github.com/potassco/clingo/archive/v5.2.1.tar.gz"
+  sha256 "b486b0adbd0a2acdfe55856036039318844ccb1e3510f2796caf03c5ee455378"
+
+  depends_on "bison" => :build
```
Comment (ilovezfs) [inline Formula/clingo.rb]: the system version of bison that ships with macOS isn't sufficient?

Comment (domoritz) [reply Formula/clingo.rb]: https://github.com/potassco/clingo/blob/613e390a4d127ba2ab4e58de32dbc10077b7482b/INSTALL.md#development-dependencies says it needs 2.5 but I only had 2.3.

Comment (domoritz) [reply Formula/clingo.rb]: Hmm, I tried the build and it compiles without bison installed but I wonder what happens if I don't add it.

**Comment by domoritz** [comment]
Ready

**Comment by domoritz** [comment]
Thank you @JCount!

File: Formula/clingo.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (domoritz) [reply Formula/clingo.rb]: Excellent. So I can remove `re2c` as well. 

The way the python module is installed seems fine to me. Thanks for writing this up!

**Comment by domoritz** [comment]
I removed `re2c` as it is not needed. https://github.com/Homebrew/homebrew-core/pull/20281#r148953423

**Comment by JCount** [comment]
@domoritz I made the last change because the current configuration had python enabled and built into the executables, but the python module itself wasn't actually getting installed into the prefix.

**Comment by domoritz** [comment]
Thanks @JCount! How does the depreciation of the existing clasp and gringo formulas work?

**Comment by domoritz** [comment]
I updated to the latest clingo 5.2.2. This PR is ready to go.

**Comment by JCount** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Review by JCount — APPROVED** [review]

**Comment by JCount** [comment]
@domoritz Thank you for this, and thank you for your patience. 😄 🎊

**Comment by DomT4** [comment]
We need a `link_overwrite` in here somewhere:
```
==> Finishing up
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/clasp
Target /usr/local/bin/clasp
is a symlink belonging to clasp. You can unlink it:
  brew unlink clasp

To force the link and overwrite all conflicting files:
  brew link --overwrite clingo

To list all files that would be deleted:
  brew link --overwrite --dry-run clingo

Possible conflicting files are:
/usr/local/bin/clasp -> /usr/local/Cellar/clasp/3.3.2/bin/clasp
/usr/local/bin/clingo -> /usr/local/Cellar/gringo/4.5.4/bin/clingo
/usr/local/bin/gringo -> /usr/local/Cellar/gringo/4.5.4/bin/gringo
/usr/local/bin/lpconvert -> /usr/local/Cellar/clasp/3.3.2/bin/lpconvert
/usr/local/bin/reify -> /usr/local/Cellar/gringo/4.5.4/bin/reify
```

**Comment by DomT4** [comment]
https://github.com/Homebrew/homebrew-core/pull/21073

**Comment by specious** [comment]
I was checking `brew doctor` and it told me I've got an unlinked `clingo`. I tried to link it:
```
$ brew link clingo
Linking /usr/local/Cellar/clingo/5.2.2...
Error: Could not symlink bin/clasp
Target /usr/local/bin/clasp
is a symlink belonging to clasp. You can unlink it:
  brew unlink clasp
```
Okay, so I tried to solve that:
```
$ brew link --overwrite clingo
Linking /usr/local/Cellar/clingo/5.2.2... 13 symlinks created

$ brew unlink clasp
Unlinking /usr/local/Cellar/clingo/5.2.2... 13 symlinks removed

$ brew link clingo
Linking /usr/local/Cellar/clingo/5.2.2... 13 symlinks created
```
Turned out `clasp` **is** `clingo`!
```
$ brew info clasp | head -1
clingo: stable 5.2.2 (bottled)
```
I can't uninstall it. I can't migrate it:
```
$ brew migrate clasp
Error: clingo doesn't replace any formula.
```
I still have `clasp` installed in my Cellar:
```
$ ls /usr/local/Cellar/clasp
3.2.0
```
Any advice for how to proceed with resolving this?

**Comment by domoritz** [comment]
You are right that clasp is not needed anymore and you only need clingo. Can you try to uninstall clasp and then install the clingo package?

**Comment by specious** [comment]
I tried that. It'll be uninstalling `clingo`:
```
$ brew uninstall clasp
Error: Refusing to uninstall /usr/local/Cellar/clingo/5.2.2
because it is required by opam 1.2.2_3, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies clasp
```
What I don't understand is what makes `brew` treat `clasp` as an alias for `clingo`?

**Comment by domoritz** [comment]
Try to uninstall opam as well and reinstall it later.

**Comment by ilovezfs** [comment]
```
Josephs-MacBook-Pro:~ joe$ ls -l /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/clasp 
lrwxr-xr-x  1 joe  admin  20 Jan  1 22:34 /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/clasp -> ../Formula/clingo.rb
Josephs-MacBook-Pro:~ joe$ 
```

```
cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
rm Aliases/clasp
brew uninstall -f --ignore-dependencies clasp
brew uninstall -f --ignore-dependencies clingo
brew prune
git checkout -- Aliases/clasp
brew install clingo
```

**Comment by specious** [comment]
@ilovezfs - unambiguously well-advised, worked like a charm 👍

**Comment by domoritz** [comment]
Excellent. Enjoy your clean homebrew!

**Pull request closed without merging**